osimpipeline/task.py

```diff
@@ -106,82 +110,59 @@ def __init__(self, trial):
         self.trial = trial
 
 class SetupTask(TrialTask):
-    def __init__(self, tool, trial, 
-                 model_to_adjust=None, 
-                 adjusted_model=None,
-                 gait_cycles='concatenated'):
+    def __init__(self, tool, trial, cycle=None):
         super(SetupTask, self).__init__(trial)
         self.tool = tool
         self.trial = trial
-        self.name = '%s_%s_setup' % (trial.id, self.tool)
         self.tool_path = os.path.join(trial.results_exp_path, self.tool)
         self.doit_path = self.study.config['doit_path']
-        self.gait_cycles=gait_cycles
-        self.source_extloads_fpath = os.path.join(self.trial.rel_path, 
-                self.tool, 'external_loads.xml')
-        self.results_extloads_fpath = os.path.join(self.tool_path,
-                os.path.basename(self.source_extloads_fpath))
-        self.source_tasks_fpath = os.path.join(self.trial.rel_path, self.tool,
-                'tasks.xml')
-        self.results_tasks_fpath = os.path.join(self.tool_path, 
-                os.path.basename(self.source_tasks_fpath))
-
-        # Specify model to be adjusted by RRA
-        if not model_to_adjust:
-             self.model_to_adjust = '%s.osim' % self.subject.name
-        else:
-            self.model_to_adjust = model_to_adjust
-
-        self.model_to_adjust_fpath = os.path.join(
-            self.subject.results_exp_path, self.model_to_adjust)
-
-        # Specify adjusted model
-        if not adjusted_model:
-            self.adjusted_model = '%s_adjusted.osim' % self.subject.name
+        self.cycle = cycle
+
+        # Generate setup file based on whether or not a specific cycle has been
+        # specified
+        if self.cycle:
+            self.create_setup_deps = (True if self.cycle.name=='cycle01' else
```

Comment:
Turns out we still need for it task dependencies. Otherwise, two different cycle tasks will have the same source file target dependency (e.g. external_loads.xml)